### PR TITLE
120 improve output

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: cli/src/main/scala/aqua/AquaError.scala [hunk 1]
```diff
@@ -10,13 +10,14 @@ sealed trait AquaError {
   def showForConsole(script: String): String
 }
 
-case class CustomSyntaxError(span: Span, message: String) extends AquaError {
+case class CustomSyntaxError(name: String, span: Span, message: String) extends AquaError {
```
Comment (alari) [inline cli/src/main/scala/aqua/AquaError.scala:13]: Here you added filename-awareness to a non-filename-aware error type: `Span.F` can be applied to any text input, and errors capture only the offset, `FileSpan.F` can be used only for file input, and augments `Span` with the filename.

I'm not sure it's a good design.

**Review by alari — CHANGES_REQUESTED** [review]

**Comment by DieMyst** [comment]
fixes #120

File: cli/src/main/scala/aqua/AquaCompiler.scala [hunk 2]
```diff
@@ -18,19 +19,30 @@ import cats.syntax.monoid._
...
 
     def hasOutput(target: CompileTarget): Boolean = target match {
       case _ => model.funcs.nonEmpty
     }
+
+    def targetPath(ext: String): Validated[Throwable, Path] =
+      Validated.catchNonFatal {
+        val fileName = srcFile.getFileName
+        if (fileName == null) {
+          throw new Exception(s"Unexpected: there is no file name in $srcFile")
```
Comment (alari) [inline cli/src/main/scala/aqua/AquaCompiler.scala:41]: You mean, `$fileName`?

File: parser/src/main/scala/aqua/parser/lift/FileSpan.scala [hunk 3]
```diff
@@ -1,10 +1,11 @@
 package aqua.parser.lift
 
-import cats.{Comonad, Eval}
 import cats.parse.{LocationMap, Parser => P}
+import cats.{Comonad, Eval}
 
 import scala.language.implicitConversions
 
+// TODO: rewrite FileSpan and Span under one trait
```
Comment (alari) [inline parser/src/main/scala/aqua/parser/lift/FileSpan.scala:8]: Why?

**Review by alari — APPROVED** [review]

File: cli/src/main/scala/aqua/AquaCompiler.scala (hunk 2 above)
Comment (DieMyst) [reply cli/src/main/scala/aqua/AquaCompiler.scala:41]: it will be null, so, no

Comment (alari) [reply cli/src/main/scala/aqua/AquaCompiler.scala:41]: I don't understand what the error message means. Please rephrase.

**Pull request merged**